ouroboros-network/src/Ouroboros/Network/AnchoredFragment.hs

```diff
@@ -464,7 +463,7 @@ AnchoredFragment a1 c1 `isPrefixOf` AnchoredFragment a2 c2 =
 -- * @join before after  == Just f@
 splitAfterPoint
    :: forall block1 block2.
-      (HasHeader block1, HasHeader block2, HeaderHash block1 ~ HeaderHash block2)
+      (HasHeader block1, HeaderHash block1 ~ HeaderHash block2)
```

Comment:
I'm in general in favour of removing redundant constraints and having the warning enabled. However, there's an argument to be made for adding constraints that should hold, even though they are unused. For consistency alone, I wouldn't mind keeping the constraint, but that would of course trigger the warning.

Reply:
I agree that is a point to keeping some redundant constraint, `HasInitiator appType ~ True` is another good example.

The PR is just a draft and I'll see what the rest of the network team thinks about the change.

Reply:
There's already asymmetry in argument's types: passing `Anochored block1` is a stronger assumption than passing just a `Point block2`. I am fine with making the type more general, though it's rather hard to imagine it will be useful (but it will constraint what one will be able to do when refactoring).

Reply:
We could have the warning on by default for all modules, and turn it off on a case by case basis in some modules where there's good reason.

This issue isn't so critical, and our current default is the other way around (off by default). How about that?

Reply:
@karknu sounds like we have consensus. Lets do it and get this PR off the TODO list.

Reply:
Ok, I'll will rebase and mark ready for review.